### PR TITLE
feat!: Add support for multi-select Custom Properties

### LINKED ISSUE #3198: Bug: Multi-select Custom Properties are not supported
## Issue

The Github API supports organizational custom properties with the following types: `string, single_select, multi_select, true_false`. Creating a property with type `multi_select` or `true_false` breaks many of the library interactions due to unmarshalling errors

API Reference: https://docs.github.com/en/rest/orgs/custom-properties?apiVersion=2022-11-28#create-or-update-custom-properties-for-an-organization

## Reproducing

- Create a custom property of type `multi_select` via the UI
- Use library to get organizational custom properties
  - Unmarshalling error
- Assign the custom property a value on a repository
-  Use the library to get the Repository
  - Unmarshalling error

### PR DESCRIPTION
**Pull request by arymoraes**
Fixes #3198.

BREAKING CHANGE: `CustomPropertyValue.Value` is changed from `*string` to `interface{}` to support `string` and `[]string` values.

So the problem here is that the GitHub custom properties `value` can come in either as a `string` (for all non `multi_select` properties), or as a `[]string` in a `multi_select` property.

The issue is happening because we were trying to unmarshal it but always expecting a `string` for the Value.

My solution was to make value an `interface{}` type that implements a `UnmarshalJSON` to handle both scenarios. While this works, we end up losing the auto generated `GetValue()` accessor. Another thing I was unsure is that I saw on other places and on the tests for this that we use a pointer to the value, but I changed it to the value itself on the Unmarshal call since we lose the accessor, not sure if this is correct.

Another solution I played with was using `json.RawMessage`, I could not make it fully work (I can give it another try), but at least we did lose the auto generated `GetValue` accessor there, would this be a more adequate solution?

### CODE REVIEW

**Comment by gmlewis** [comment]
Yes, if I remember correctly, we handle situations like this in various ways:

* with `interface{}` like you did (e.g. [copilot](https://github.com/google/go-github/blob/dd756703cb6b93cc8dc3e826799128c4e8707db6/github/copilot.go#L47) )
* with `json.RawMessage` like you mention (e.g. [scim](https://github.com/google/go-github/blob/dd756703cb6b93cc8dc3e826799128c4e8707db6/github/scim.go#L181) , [repos_deployments](https://github.com/google/go-github/blob/dd756703cb6b93cc8dc3e826799128c4e8707db6/github/repos_deployments.go#L22) , [repos_rules](https://github.com/google/go-github/blob/dd756703cb6b93cc8dc3e826799128c4e8707db6/github/repos_rules.go#L129) , etc.)
* with multiple structs (not finding any good examples at the moment)

So I'm fine with this method. You could additionally define two new structs `CustomPropertyValueWithString` and `CustomPropertyValueWithStringSlice` that have the respective types for `Value` and then two helper functions could each return one of the new types with a `bool` (indicating if the returned pointer is actually that type or not) if you want... but I'm fine if you want to just run with the current solution too.

**Comment by arymoraes** [comment]
> Yes, if I remember correctly, we handle situations like this in various ways:
> 
> * with `interface{}` like you did (e.g. [copilot](https://github.com/google/go-github/blob/dd756703cb6b93cc8dc3e826799128c4e8707db6/github/copilot.go#L47) )
> * with `json.RawMessage` like you mention (e.g. [scim](https://github.com/google/go-github/blob/dd756703cb6b93cc8dc3e826799128c4e8707db6/github/scim.go#L181) , [repos_deployments](https://github.com/google/go-github/blob/dd756703cb6b93cc8dc3e826799128c4e8707db6/github/repos_deployments.go#L22) , [repos_rules](https://github.com/google/go-github/blob/dd756703cb6b93cc8dc3e826799128c4e8707db6/github/repos_rules.go#L129) , etc.)
> * with multiple structs (not finding any good examples at the moment)
> 
> So I'm fine with this method. You could additionally define two new structs `CustomPropertyValueWithString` and `CustomPropertyValueWithStringSlice` that have the respective types for `Value` and then two helper functions could each return one of the new types with a `bool` (indicating if the returned pointer is actually that type or not) if you want... but I'm fine if you want to just run with the current solution too.

That makes sense, thank you for the explanation and the breakdown. I'm fine with the current solution as is if we want to go with that, at some point I did something similar to having two new structs but ended up deciding to move back to trying the `interface{}` approach.

**Review by gmlewis — APPROVED** [review]
Thank you, @arymoraes!
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Review by valbeat — APPROVED** [review]

**Comment by gmlewis** [comment]
Thank you, @valbeat !
Merging.

File: github/orgs_properties.go
Comment (darend) [inline github/orgs_properties.go:64]: Do boolean custom properties come through as a string as well?

https://docs.github.com/en/organizations/managing-organization-settings/managing-custom-properties-for-repositories-in-your-organization#adding-custom-properties

Comment (arymoraes) [reply github/orgs_properties.go:64]: Yes, they come as `string` from the GitHub API:

```json
      {
        "property_name": "boolean_prop",
        "value": "true"
      },
```

**Pull request merged**
